### DISCUSSION TITLE
Do we have graphql playground in directUs? if yes how to access it.

### DISCUSSION THREAD

**Discussion by alwaz-shahid** [directus/directus#9863] [Archive]
I'm looking forward to use graphql with directus but there is no graphql playground and writing all those queries manually can be a nightmare.

What's the future scope of directus with graphql?

**Comment by rijkvanzanten** [comment]
Hi there! Directus doesn't ship with any devtooling in its production bundle by default. This includes REST API tools like postman or graphql explorers like graphiql. There's plenty of third party graphql explorers that are compatible with directus, like graphiql, graphiql-online, paw, altair, etc etc 👍🏻

**Reply by SalahAdDin** [reply]
But, what's about development? I'm trying to get it working in local with **docker**.

**Reply by br41nslug** [reply]
I personally use this online graphql playground when developing https://lucasconstantino.github.io/graphiql-online/ works like a charm for local docker development without the need to setup/host a separate app but there are plenty of others as Rijk mentioned.

**Reply by rijkvanzanten** [reply]
I'll copy here what I already answered on #5296 as well: 

> > Why?
> 
> Because there's a lot of different options available, and everybody has their own preference as to which one is the best tool for the job. We shipped with GraphiQL for a little bit, and people immediately started asking for graphql-playground instead. At the end of the day, there's no real benefit to hosting our own copy of those popular tools, as they all work perfectly as standalone apps (which also has the benefit you can use the same tool for multiple APIs). Even since may 10th last year there's a couple new popular ones that were added, including Insomnia and Postman

**Reply by benhaynes** [reply]
I think the solution would be to create 1+ extensions (Modules) that offers this. Then it's easy to include, but not opinionated within Core. Just need the dev experience for extensions to get cleaned up first.

**Reply by SalahAdDin** [reply]
> I'll copy here what I already answered on #5296 as well:
> 
> > > Why?
> > 
> > 
> > Because there are a lot of different options available, and everybody has their own preference as to which one is the best tool for the job. We shipped with GraphiQL for a little bit, and people immediately started asking for graphql-playground instead. At the end of the day, there's no real benefit to hosting our own copy of those popular tools, as they all work perfectly as standalone apps (which also has the benefit you can use the same tool for multiple APIs). Even since May 10th last year there's a couple of new popular ones that were added, including Insomnia and Postman

Not for hosting, just for local development.

> I think the solution would be to create 1+ extensions (Modules) that offers this. Then it's easy to include, but not opinionated within Core. Just need the dev experience for extensions to get cleaned up first.

That's a good idea.
